apps/schema/src/components/SchemaApp.vue

```diff
@@ -0,0 +1,292 @@
+            Save
+          </ButtonAction>
+          <ButtonAction v-if="dirty" @click="loadSchema" class="ml-2">
+            Reset
+          </ButtonAction>
+          <ButtonAction
+            v-if="schema.tables?.length > 0"
+            @click="toggleShowDiagram"
+            class="ml-2"
+          >
+            {{ showDiagram ? "Hide" : "Show" }} Diagram
+          </ButtonAction>
+          <MessageError v-if="error" class="ml-2 m-0 p-2">
+            {{ error }}
+          </MessageError>
+          <MessageWarning v-if="warning" class="ml-2 m-0 p-2">
+            {{ warning }}
+          </MessageWarning>
+          <MessageSuccess v-if="success" class="ml-2 m-0 p-2">
+            {{ success }}
+          </MessageSuccess>
+        </div>
+      </div>
+    </div>
+    <Spinner v-if="loading === true" />
+    <div v-else class="row">
+      <div class="col-2 bg-white">
+        <SchemaToc
+          v-model="schema"
+          v-if="schema.tables"
+          @input="handleInput"
+          :key="key"
+          :isManager="isManager"
+        />
+      </div>
+      <div class="bg-white col ml-2 overflow-auto">
+        <a id="molgenis_diagram_anchor"></a>
+        <NomnomDiagram
+          :schema="schema"
+          :key="JSON.stringify(schema)"
+          v-if="showDiagram"
+        />
+        <SchemaView
+          v-model="schema"
+          :schemaNames="schemaNames"
+          @input="handleInput"
+          :isManager="isManager"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<style scoped>
+table {
+  table-layout: fixed;
+}
+</style>
+
+<script>
+import { request } from "graphql-request";
+import SchemaView from "./SchemaView.vue";
+import SchemaToc from "./SchemaToc.vue";
+import NomnomDiagram from "./NomnomDiagram.vue";
+import {
+  ButtonAction,
+  MessageError,
+  MessageSuccess,
+  MessageWarning,
+  Spinner,
+} from "molgenis-components";
+
+export default {
+  components: {
+    SchemaView,
+    ButtonAction,
+    MessageError,
+    MessageWarning,
+    MessageSuccess,
+    SchemaToc,
+    Spinner,
+    NomnomDiagram,
+  },
+  data() {
+    return {
+      schema: {},
+      loading: false,
+      error: null,
+      warning: null,
+      success: null,
+      showDiagram: false,
+      schemaNames: [],
+      dirty: false,
+      key: Date.now(),
+      isManager: false,
+    };
+  },
+  methods: {
+    handleInput() {
+      this.dirty = true;
+      this.success = null;
+      this.key = Date.now();
+      this.$emit("input", this.schema);
+    },
+    toggleShowDiagram() {
+      this.showDiagram = !this.showDiagram;
+      this.$scrollTo({ el: "#molgenis_diagram_anchor", offset: -50 });
+    },
+    saveSchema() {
+      this.loading = true;
+      this.graphqlError = null;
+      this.warning = "submitting changes";
+      this.success = null;
+      //copy so in case of error user can continue to edit
+      let schema = JSON.parse(JSON.stringify(this.schema));
```

Comment:
use clone function instead of rewriting it

Reply:
I am not a big fan of creating unnecessary dependencies on other libraries when the code is soo little. I find that a bit too 'dry'. But if your eyes start bleeding I don't mind to change.

Reply:
ok, found out how to do it.